test/ssl.jl

```diff
@@ -0,0 +1,109 @@
+function ssl_tests(implicit::Bool = true)
+    mode = implicit ? :implicit : :explicit
+
+    setup_server()
+
+    server = FTPServer(security=mode)
+
+    opts = (
+        :hostname => hostname(server),
+        :username => username(server),
+        :password => password(server),
+        :ssl => true,
+        :implicit => implicit,
+        :verify_peer => false,
+    )
+    try
+        options = RequestOptions(; opts..., active_mode=false)
+        test_download(options)
+        test_upload(options)
+        test_cmd(options)
+
+        options = RequestOptions(; opts..., active_mode=true)
+        test_download(options)
+        test_upload(options)
+        test_cmd(options)
+
+        options = RequestOptions(; opts..., active_mode=false)
+        ctxt, resp = ftp_connect(options)
+        @test resp.code == 226
+
+        test_cmd(ctxt)
+        test_download(ctxt)
+        test_upload(ctxt)
+
+        options = RequestOptions(; opts..., active_mode=true)
+        ctxt, resp = ftp_connect(options)
+        @test resp.code == 226
+
+        test_cmd(ctxt)
+        test_download(ctxt)
+        test_upload(ctxt)
+
+        # the connection has to be closed during test_upload(ctxt) to get the server file to write out
+        # ftp_close_connection(ctxt)
+    finally
+        close(server)
+
+        for line in eachline(server.io)
+            print(line)
+            if contains(line, "FTP session closed")
+                break
+            end
+        end
```

Comment:
Actually the output from the server isn't really that useful, so I think I'm going to just remove it.